src/iuse.cpp

```diff
@@ -7490,48 +7490,80 @@ int iuse::ladder( player *p, item *, bool, const tripoint& )
     return 1;
 }
 
+
 int iuse::washclothes( player *p, item *it, bool, const tripoint& )
 {
+    // Check that player isn't over volume limit as this might cause it to break... this is a hack.
+    // TODO: find a better solution.
+    if (p->volume_capacity() < p->volume_carried()){
+        p->add_msg_if_player( _( "You're carrying too much to clean anything." ) );
+        return 0;
+    }
     if( it->charges < it->type->charges_to_use() ) {
         p->add_msg_if_player( _( "You need a cleansing agent to use this." ) );
         return 0;
     }
 
-    const int pos = g->inv_for_flag( "FILTHY", _( "Wash what?" ) );
-    item &mod = p->i_at( pos );
-    if( pos == INT_MIN ) {
-        p->add_msg_if_player( m_info, _( "Never mind." ) );
+
+
+    player player = *p;
+    
+    player.inv.restack( player );
+
+    const inventory_filter_preset preset( []( const item_location & location ) {
+        return (*location).item_tags.find("FILTHY") != (*location).item_tags.end();
+    } );
+    // Maybe this should also search surrounding area, not just player inventory?
+    inventory_iuse_selector inv_s( player, _("ITEMS TO CLEAN"), preset );
+    inv_s.add_character_items( player );
+    inv_s.set_title( _( "Multiclean" ) );
+    inv_s.set_hint( _( "To clean x items, type a number before selecting." ) );
+    std::list<std::pair<int, int>> to_clean;
+    if( inv_s.empty() ) {
+        popup( std::string( _( "You have nothing to clean." ) ), PF_GET_KEY );
+        to_clean = std::list<std::pair<int, int> >();
         return 0;
     }
 
-    const int required_water = 2 * mod.volume() / 250_ml;
-    const int time = 1000 * mod.volume() / 250_ml;
-    int required_cleanser = mod.volume() / 1000_ml;
+    to_clean = inv_s.execute();
+    if (to_clean.size() == 0){
+        return 0;
+    }
+    std::list<std::pair<int, int>>::iterator pairs;
+    int required_water = 0;
+    int time = 0;
+    int required_cleanser = 0;
 
-    if( required_cleanser < 1 ) {
+    // Determine if we have enough water and cleanser for all the items.
+    for (pairs = to_clean.begin(); pairs != to_clean.end(); ++pairs){
```

Comment:
This (and the similar loop below) should also be range-based loops. There is no need to declare the loop variable outside of the loop when it's not used there.